### PR TITLE
Consolidate daily workflows into one workflow and add email notifs

### PR DESCRIPTION
**Pull request by pauladkisson**
This PR consolidates the daily workflows into dailies.yml and adds an email notif for each to make it easier to keep track of them all.

See https://github.com/catalystneuro/neuroconv/actions/runs/10889811236

### CODE REVIEW

File: .github/workflows/update-s3-testing-data.yml
Comment (h-mayorquin) [inline .github/workflows/update-s3-testing-data.yml:1]: This one is diabled right now, let's keep it out of the dallies and handle caching of the test separately.

Comment (pauladkisson) [reply .github/workflows/update-s3-testing-data.yml:1]: Why is it disabled? just curious

**Review by h-mayorquin — COMMENTED** [review]
This LGTM and it would make your work easier, right?

 The only thing I would keep out is the s3 thing.

**Comment by pauladkisson** [comment]
> This LGTM and it would make your work easier, right?

Yes, definitely.

File: .github/workflows/update-s3-testing-data.yml
Comment (h-mayorquin) [reply .github/workflows/update-s3-testing-data.yml:1]: I think @CodyCBakerPhD would be able to give you a better summary. I think it was just brittle and cranky. So far the workflow has been to upload the files directly to S3.

I have been running a multi-os gin testing framework in Spikeinterface without s3 without trouble for a while so maybe we could move in that direction but no need to rush anything:

https://github.com/SpikeInterface/spikeinterface/blob/main/.github/workflows/caches_cron_job.yml

Comment (pauladkisson) [reply .github/workflows/update-s3-testing-data.yml:1]: Yeah, and we've been using it in the dailies for roiextractors for a while: https://github.com/catalystneuro/roiextractors/blob/main/.github/workflows/dailies.yaml

But I can leave it out for now.

Comment (h-mayorquin) [reply .github/workflows/update-s3-testing-data.yml:1]: Ah, if that is causing no hitch there and that has not shown in s3 costs here I guess I don't mind. Approving.

**Review by h-mayorquin — APPROVED** [review]

**Comment by pauladkisson** [comment]
Looks like the daily docker tests have been failing for a while now: https://github.com/catalystneuro/neuroconv/actions/runs/10855171853/job/30127190657

Want to fix before merging this.

**Comment by h-mayorquin** [comment]
Ok. Looks like a credentials issue.

**Comment by pauladkisson** [comment]
Ok, daily docker tests is passing now!

https://github.com/catalystneuro/neuroconv/actions/runs/10858022361/job/30135424576

**Comment by pauladkisson** [comment]
build-and-upload-docker-image-dev still failing tho: https://github.com/catalystneuro/neuroconv/actions/runs/10855171853/job/30127196383

**Pull request merged**